services/bundle_analysis/new_notify/contexts/commit_status.py

```diff
@@ -0,0 +1,185 @@
+                the BundleAnalysisComparison.
+        """
+        if self.is_field_loaded("bundle_analysis_comparison"):
+            return self
+        pull = self._notification_context.pull
+        try:
+            if pull is None:
+                comparison = ComparisonLoader(
+                    base_commit=self._notification_context.commit.get_parent_commit(),
+                    head_commit=self._notification_context.commit,
+                ).get_comparison()
+            else:
+                comparison = ComparisonLoader.from_EnrichedPull(pull).get_comparison()
+            self._notification_context.bundle_analysis_comparison = comparison
+            return self
+        except (
+            MissingBaseCommit,
+            MissingHeadCommit,
+            MissingBaseReport,
+            MissingHeadReport,
+        ) as exp:
+            raise NotificationContextBuildError(
+                "load_bundle_comparison", detail=exp.__class__.__name__
+            )
+
+    def load_commit_status_level(self) -> "CommitStatusNotificationContextBuilder":
+        bundle_analysis_comparison = (
+            self._notification_context.bundle_analysis_comparison
+        )
+        user_config = self._notification_context.user_config
+
+        if is_bundle_change_within_bundle_threshold(
+            bundle_analysis_comparison, user_config.warning_threshold
+        ):
+            self._notification_context.commit_status_level = CommitStatusLevel.INFO
+        elif user_config.status_level == "informational":
+            self._notification_context.commit_status_level = CommitStatusLevel.WARNING
+        else:
+            self._notification_context.commit_status_level = CommitStatusLevel.ERROR
+        return self
+
+    def load_commit_status_url(self) -> "CommitStatusNotificationContextBuilder":
+        if self._notification_context.pull:
+            self._notification_context.commit_status_url = get_bundle_analysis_pull_url(
+                self._notification_context.pull.database_pull
+            )
+        else:
+            self._notification_context.commit_status_url = get_commit_url(
+                self._notification_context.commit
+            )
+        return self
+
+    def load_cache_ttl(self) -> "CommitStatusNotificationContextBuilder":
+        self._notification_context.cache_ttl = int(
+            # using `get_config` instead of `current_yaml` because
+            # `current_yaml` does not include the install configuration
+            get_config("setup", "cache", "send_status_notification", default=600)
+        )  # 10 min default
+        return self
+
+    def build_context(self) -> "CommitStatusNotificationContextBuilder":
+        super().build_context()
+        async_to_sync(self.load_optional_enriched_pull)()
+        self.load_bundle_comparison()
+        self.load_commit_status_level()
+        self.load_commit_status_url()
+        self.load_cache_ttl()
+        return self
```

Comment:
```suggestion
        return (self.load_bundle_comparison()
            .load_commit_status_level()
            .load_commit_status_url()
            .load_cache_ttl())
```

Since you made all these methods return `Self` :-)